oak-query-spi/src/main/java/org/apache/jackrabbit/oak/spi/query/package-info.java

```diff
@@ -18,7 +18,7 @@
 /**
  * This package contains oak query index related classes.
  */
-@Version("2.0.0")
+@Version("3.0.0")
```

Comment:
why a major version bump ? There doesn't seem to be any breaking change included in this PR.

Reply:
I believe this is a non-internal API, so we need to be extra careful here.

Reply:
> I believe this is a non-internal API, so we need to be extra careful here.

It is supposed to be an internal API. I'm not aware of any external users.

We (relatively) recently added getIgnoredClassNamesInCallTrace which also caused a version bump, if I remember correctly. This didn't cause any issues.

Reply:
If it is internal, it should be marked as that ("@Internal") annotation.

Reply:
You can't do that. It is not internal. Once it's public it stays public.